Sources/StreamChatUI/Utils/ImageLoading/ImageLoading.swift

```diff
@@ -2,31 +2,60 @@
 // Copyright © 2022 Stream.io Inc. All rights reserved.
 //
 
+import StreamChat
 import UIKit
 
-/// ImageLoading is providing set of functions for downloading of images from URLs.
+/// A protocol that provides a set of functions for loading images.
 public protocol ImageLoading: AnyObject {
-    /// Load an image from using the given URL request
+    /// Load an image into an imageView from the given `URL`.
     /// - Parameters:
-    ///   - urlRequest: The `URLRequest` object used to fetch the image
-    ///   - cachingKey: The key to be used for caching this image
-    ///   - completion: Completion that gets called when the download is finished
+    ///   - imageView: The image view where the image will be loaded.
+    ///   - url: The `URL` of the image. If `nil` it will load the placeholder.
+    ///   - options: The loading options on how to fetch the image.
+    ///   - completion: The completion when the loading is finished.
+    /// - Returns: A cancellable task.
+    @discardableResult
+    func loadImage(
+        into imageView: UIImageView,
+        from url: URL?,
+        with options: ImageLoaderOptions,
+        completion: ((_ result: Result<UIImage, Error>) -> Void)?
+    ) -> Cancellable?
+
+    /// Download an image from the given `URL`.
+    /// - Parameters:
+    ///   - url: The `URL` of the image.
+    ///   - options: The loading options on how to fetch the image.
+    ///   - completion: The completion when the loading is finished.
+    /// - Returns: A cancellable task.
+    @discardableResult
+    func downloadImage(
+        from url: URL,
+        with options: ImageDownloadOptions,
+        completion: @escaping ((_ result: Result<UIImage, Error>) -> Void)
+    ) -> Cancellable?
+
+    /// Load a batch of images and get notified when all of them complete loading.
+    /// - Parameters:
+    ///   - urlsAndOptions: A tuple of urls and the options on how to fetch the image.
+    ///   - completion: The completion when the loading is finished.
+    ///   It returns an array of image and errors in case the image failed to load.
+    func downloadMultipleImages(
+        from urlsAndOptions: [(url: URL, options: ImageDownloadOptions)],
```

Comment:
The problem then is that you don't guarantee that you have an option per URL, that is why here a tuple makes sense. But lets the opinion of Pol as well 👍

Reply:
I don't get your point @nuno-vieira , I think it is better to have an struct that has url and options, and pass here an array of those objects

Reply:
@polqf @martinmitrevski What should I name this struct? If I create this data structure, it probably makes sense to use it as well in the other functions as well. In the other functions of the `ImageLoading`, we pass an URL and Options as separate arguments. If because of this API we are making them as 1 structure, then for consistency it would make sense to use this everywhere 🤔 

But then, what is the name we should give? Some ideas:
- `ImageRemoteResource`
- `ImageDownloadResource`
- `ImageDownload`
- `ImageDownloadInfo`

Reply:
I would go with `ImageDownloadInfo`, but no strong opinion here. And yes, ideally it should be consistent throughout the API.

Reply:
For ios `Info` is usually used for dictionaries 🤔 What about `ImageDownloadItem`?

Reply:
We already use `Info` in several places in the SDK, such as `UserInfo`, so I don't think it's a problem. However, as mentioned, no strong preference, can also be `ImageDownloadItem`.

Reply:
I think now it is much better: https://github.com/GetStream/stream-chat-swift/pull/2339/commits/1802b0bdf3ec6f59ee42c074d7e0e6119c2b1c60

🙂

Reply:
it's much better indeed 👍